src/frontend/planner_test/tests/testdata/cast.yaml

```diff
@@ -0,0 +1,90 @@
+  sql: |
+    select 1 where NULL;
+  batch_plan: |
+    BatchProject { exprs: [1:Int32] }
+    └─BatchFilter { predicate: null:Boolean }
+      └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (CASE NULL)
+  sql: |
+    select case when NULL then 1 end;
+  batch_plan: |
+    BatchProject { exprs: [Case(null:Boolean, 1:Int32)] }
+    └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (JOIN ON NULL)
+  sql: "create table t1(v1 int); \ncreate table t2(v1 int); \nselect t1.v1 from t1 join t2 on NULL;\n"
+  batch_plan: |
+    BatchNestedLoopJoin { type: Inner, predicate: true, output: all }
+    ├─BatchExchange { order: [], dist: Single }
+    | └─BatchFilter { predicate: null:Boolean }
+    |   └─BatchScan { table: t1, columns: [t1.v1], distribution: SomeShard }
+    └─BatchExchange { order: [], dist: Single }
+      └─BatchFilter { predicate: null:Boolean }
+        └─BatchScan { table: t2, columns: [], distribution: SomeShard }
+- name: current_schemas (current_schemas(NULL))
+  sql: |
+    select current_schemas(NULL);
+- name: FILTER (FILTER NULL)
+  sql: |
+    create table t(v1 int);
+    select count(*) FILTER(WHERE null) from t;
+  batch_plan: |
+    BatchSimpleAgg { aggs: [sum0(count filter(null:Boolean))] }
+    └─BatchExchange { order: [], dist: Single }
+      └─BatchSimpleAgg { aggs: [count filter(null:Boolean)] }
+        └─BatchScan { table: t, columns: [], distribution: SomeShard }
+- name: implicit cast boolean (HAVING VARCHAR)
+  sql: |
+    select 1 having 'y';
+  batch_plan: |
+    BatchProject { exprs: [1:Int32] }
+    └─BatchFilter { predicate: 'y':Varchar::Boolean }
+      └─BatchSimpleAgg { aggs: [] }
+        └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (WHERE VARCHAR)
+  sql: |
+    select 1 where 'y';
+  batch_plan: |
+    BatchProject { exprs: [1:Int32] }
+    └─BatchFilter { predicate: 'y':Varchar::Boolean }
+      └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (CASE VARCHAR)
+  sql: |
+    select case when 'y' then 1 end;
+  batch_plan: |
+    BatchProject { exprs: [Case('y':Varchar::Boolean, 1:Int32)] }
+    └─BatchValues { rows: [[]] }
+- name: implicit cast boolean (JOIN ON VARCHAR)
+  sql: "create table t1(v1 int); \ncreate table t2(v1 int); \nselect t1.v1 from t1 join t2 on 'y';\n"
+  batch_plan: |
+    BatchNestedLoopJoin { type: Inner, predicate: true, output: all }
+    ├─BatchExchange { order: [], dist: Single }
+    | └─BatchFilter { predicate: 'y':Varchar::Boolean }
+    |   └─BatchScan { table: t1, columns: [t1.v1], distribution: SomeShard }
+    └─BatchExchange { order: [], dist: Single }
+      └─BatchFilter { predicate: 'y':Varchar::Boolean }
+        └─BatchScan { table: t2, columns: [], distribution: SomeShard }
+- name: current_schemas (current_schemas(VARCHAR))
+  sql: |
+    select current_schemas('y');
+  binder_error: |-
+    Feature is not yet implemented: Only boolean literals are supported in `current_schemas`.
```

Comment:
See: https://github.com/risingwavelabs/risingwave/pull/7325/files#r1067114884

Reply:
This will be solved once we generalize another place from handling `null` to `unknown` types: #7320